### ISSUE TITLE
Build hangs on eql5 make

### ISSUE THREAD

**Issue by deadtrickster** [nEXT-Browser/nEXT#1]
`eql5 -qgui` works!

```
~/Downloads/nEXT/next$ eql5 make
EQL5 17.7.1 (ECL 16.1.3, Qt 5.7.1)
;;; Loading #P"/home/dead/.eclrc"
;;; Loading "/home/dead/quicklisp/setup.lisp"
;;; Loading #P"/usr/local/lib/ecl-16.1.3/asdf.fas"
To load "cl-strings":
  Load 1 ASDF system:
    cl-strings
; Loading "cl-strings"
```

And it's been like this for a day :-)

OS - Ubuntu 17
ECL - 16.1.3 built from tarballs

**Comment by jmercouris** [comment]
Hey! Thank you for the report! I'm still working on the build system together with the developer of EQL,  there are some issues with ASDF and packaging of the whole project. Ideally I'd like to migrate to an entirely lisp "build" system (including deployment to other OS)


In the meantime, can you please try the following:
`eql5 run.lisp` in the directory `Next/next`


After executing the above command, a window should come up, and you can test it by running "C-l" and typing a URL, followed by RET


Please let me know if this helps! If you'd like to contribute on the build system I can also loop you in on this conversation.

Thank you!

**Comment by deadtrickster** [comment]
I must've done something wrong, this is the output of `eql5 run.lisp`:

```
EQL5 17.7.1 (ECL 16.1.3, Qt 5.7.1)
;;; Loading #P"/home/dead/.eclrc"
;;; Loading "/home/dead/quicklisp/setup.lisp"
;;; Loading #P"/usr/local/lib/ecl-16.1.3/asdf.fas"
To load "cl-strings":
  Load 1 ASDF system:
    cl-strings
; Loading "cl-strings"


Condition of type: UNDEFINED-FUNCTION
The function NEXT::|setMaximumHeight| is undefined.
Available restarts:

1. (TRY-RECOMPILING) Recompile lisp/base and try loading it again
2. (RETRY) Retry loading FASL for #<cl-source-file "next" "lisp/base">.
3. (ACCEPT) Continue, treating loading FASL for #<cl-source-file "next" "lisp/base"> as having been successful.
4. (RETRY) Retry ASDF operation.
5. (CLEAR-CONFIGURATION-AND-RETRY) Retry ASDF operation after resetting the configuration.

Top level in: #<process TOP-LEVEL>.
```

`eql5 -qgui` still works though!

**Comment by jmercouris** [comment]
Hey, I'm sorry this was an omission of my own in the readme for the installation, I'll make a note of it to add these details. The function it is trying to find is one of the wrapper functions that come with EQL if you build it with them. From the EQL readme, please try the following:

> If you want to add the wrappers permanently, build EQL as usual, then run

  eql5 make-wrappers.lisp

Re-link EQL doing something like:

  touch tmp/eql.o*   (or delete "tmp/eql.o*")
  qmake eql_lib.pro
  make
  sudo make install  (Unix only)

Note: The resulting shared library should be stripped (automatically done by
"make install"); this will reduce its size considerably.

**Comment by deadtrickster** [comment]
Thanks! that helped. `eql5 run.lisp` works now, although it says "Key Undefined" when I press "Ctrl-l", "Alt-l", however, worked (i.e. minibuffer popped-up). Nor backspace/delete work.

**Comment by deadtrickster** [comment]
Ok. I figured Ctrl-l out. When window pops up and google loaded I have to click on it (i.e. on white background and only after Ctrl-l works. But this is probably another issue (or feature). Also web browser control doesn't seem to respond to resizes. Do you have your todo list shared somewhere around?

**Comment by jmercouris** [comment]
Hello! I've updated the `documents/timeline.org` which includes information about the current tasks and what I am working on. If you want to work on a task, please let me know which one so I don't work on it too. I would also be very interested in hearing any critique on the current code/suggestions!


The reason it says "Key Undefined" is because there is something called a QEventFilter, which basically allows you to capture events (in this case keystrokes) for any particular widget (hence the filter). When you set a QEventFilter with no target widget, it'll iterate through every widget calling the "KeyPressed" function for each of them (please see base.lisp), until one of them returns a Truthy value (essentially saying "I handled the keypress, no need to send to another widget).


This is problematic, we don't want to call the KeyHandler several times every time a single key is pressed. In order to get around this we could set the filtered widget, but that would mean when that particular widget is not in focus, and some other default widget of QT handles a keystroke, we may not necessarily get a signal. This is what happens when you have to click on the frame before typing in a hotkey like `C-l`.


The result is, I have set the QEventFilter to nil which calls the keypress handler multiple times. This is okay because as soon as a valid keystroke combination is consumed, the key event stack is made nil again, meaning that there will not be multiple invocations of the KeyPress action, unfortunately though the program will print multiple times "Key Undefined".


I hope that made sense :)

You can also read around a little bit in keymap.lisp, I've tried to name things logically and comment them. Thank you for help so far!
